### PR TITLE
Fix flaky test for stats because of other test inteference

### PR DESCRIPTION
**Pull request by matriv**
Added a uniqueId and use filtering on sys.jobs_log since once we enable the stats
other tests can interfere and we can have multiple entries in the jobs_log table.

### CODE REVIEW

**Comment by matriv** [comment]
@aelesbao Please review

File: sql/src/test/java/io/crate/integrationtests/PostgresITest.java [hunk 1]
```diff
@@ -305,15 +310,18 @@ public void testStatsTableFailure() throws Exception {
...
-            } catch (Exception e) {}
-
-            ResultSet resultSet = conn.prepareStatement("select stmt, error from sys.jobs_log order by ended desc;")
-                .executeQuery();
-            assertThat(resultSet.next(), is(true));
-            assertThat(resultSet.getString(1), is("insert into t(a,b) values(null, 'test')"));
-            assertThat(resultSet.getString(2), is("Cannot insert null value for column a"));
+                conn.prepareStatement(stmtStr).execute();
+            } catch (Exception e) {
+                ResultSet resultSet = conn.prepareStatement("select stmt, error from sys.jobs_log " +
```
Comment (aelesbao) [inline sql/src/test/java/io/crate/integrationtests/PostgresITest.java]: is it expected that this test will always throw and get into this catch?

Comment (matriv) [inline sql/src/test/java/io/crate/integrationtests/PostgresITest.java]: Yep, and we're asserting the error message from the jobs_log table.

Comment (aelesbao) [inline sql/src/test/java/io/crate/integrationtests/PostgresITest.java]: maybe you could add a `fail` after line 317 just to assure that the test won't pass if the error doesn't happen

**Comment by matriv** [comment]
@aelesbao  Plz check fixup

**Comment by aelesbao** [comment]
lgtm

**Pull request merged**